nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/ResultSetRecordSet.java

```diff
@@ -153,25 +153,32 @@ private static RecordSchema createSchema(final ResultSet rs, final RecordSchema
             final int column = i + 1;
             final int sqlType = metadata.getColumnType(column);
 
-            final DataType dataType = getDataType(sqlType, rs, column, readerSchema);
             final String fieldName = metadata.getColumnLabel(column);
+            Optional<RecordField> readerField = readerSchema == null ? Optional.empty() : readerSchema.getField(fieldName);
```

Comment:
I'm not too sure if the readerField Optinoal should be inlined in the if block, as it is only used in the ture case. But as it checks readerSchema available and field available this could be fine.